src/repl/scala/tools/nsc/interpreter/$iw.scala

```diff
@@ -0,0 +1,9 @@
+package scala.tools.nsc.interpreter
+
+// The REPL adds a magic import of "import scala.tools.nsc.interpreter.$iw"
+// to increase the effective nesting level of the typechecker.
+//
+// `foo` will bind to `baz.foo` in this example:
+//
+// { import bar.foo; import interpreseter.$iw; import baz.foo; foo }
+class $iw
```

Comment:
Would be nice if this has a self describing name, rather than the nod to nostalgia, but I couldn't think of a decent name when I hacked this together? 

 - `importDepthControl`
 - `increaseImportDepth`
 - `` `{` ``
 - ...